cognite_toolkit/_cdf_tk/commands/_upload.py

```diff
@@ -0,0 +1,88 @@
+from functools import partial
+from pathlib import Path
+from typing import Any
+
+from rich.console import Console
+
+from cognite_toolkit._cdf_tk.exceptions import ToolkitValueError
+from cognite_toolkit._cdf_tk.storageio import StorageIO
+from cognite_toolkit._cdf_tk.storageio._base import T_CogniteResourceList, T_StorageID, T_WritableCogniteResourceList
+from cognite_toolkit._cdf_tk.utils.collection import chunker
+from cognite_toolkit._cdf_tk.utils.fileio import FileReader
+from cognite_toolkit._cdf_tk.utils.producer_worker import ProducerWorkerExecutor
+from cognite_toolkit._cdf_tk.utils.useful_types import JsonVal
+
+from ._base import ToolkitCommand
+
+
+class UploadCommand(ToolkitCommand):
+    """Command for uploading data to CDF from files in a specified directory.
+
+    This command reads files matching a specific pattern in the input directory,
+    processes them using a StorageIO instance, and uploads the data to CDF.
+
+    Attributes:
+        _MAX_QUEUE_SIZE (int): The maximum size of the queue for processing items.
+            Set to 80 to balance memory usage and processing speed.
+    """
+
+    _MAX_QUEUE_SIZE = 80
+
+    def upload(
+        self,
+        io: StorageIO[T_StorageID, T_CogniteResourceList, T_WritableCogniteResourceList],
+        input_dir: Path,
+        ensure_configurations: bool,
+        dry_run: bool,
+        verbose: bool,
+    ) -> None:
+        """Uploads data from files in the specified input directory to CDF.
+
+        Args:
+            io: The StorageIO instance that defines how to upload the data.
+            input_dir: The directory containing the files to upload.
+            ensure_configurations: If True, creates a configuration for the upload. For example,
+                in the case of uploading RAW tables, this will create the RAW database and table.
+                For asset-centric, this will create labels and data sets.
+            dry_run: If True, performs a dry run without actually uploading the data.
+            verbose: If True, prints detailed information about the upload process.
+
+        """
+        console = Console()
+        files = list(input_dir.glob(f"*.{io.kind}.*"))
```

Comment:
How many files are you expecting to fetch here? Depending on the order of magnitude, you might face some high memory consumption here.

Reply:
For a list of files in a directory? I expect in the order of 1-10

Reply:
Ah ok. I saw it and it had some similarity with the file extractor codebase, but it's a different use case so no  issue here.